test/Conversion/TorchToTcp/data_movement.mlir

```diff
@@ -11,3 +11,17 @@ func.func @torch.aten.cat(%arg0: !torch.vtensor<[?,?],f32>, %arg1: !torch.vtenso
   %1 = torch.aten.cat %0, %int0 : !torch.list<vtensor>, !torch.int -> !torch.vtensor<[?,?],f32>
   return %1 : !torch.vtensor<[?,?],f32>
 }
+
+// CHECK-LABEL: @torch.aten.slice.Tensor
+//   CHECK-SAME:   %[[ARG0:.+]]: !torch.vtensor<[1,56,?,?],f32>
+//        CHECK:   %[[V1:.+]] = torch_c.to_builtin_tensor %[[ARG0]] : !torch.vtensor<[1,56,?,?],f32> -> tensor<1x56x?x?xf32>
+//        CHECK:   %[[V2:.+]] = tensor.extract_slice %[[V1]][%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}] [%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}] [%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}] : tensor<1x56x?x?xf32> to tensor<?x?x?x?xf32>
+//        CHECK:   %[[V3:.+]] = tensor.cast %[[V2]] : tensor<?x?x?x?xf32> to tensor<1x28x?x?xf32>
+func.func @torch.aten.slice.Tensor(%arg0: !torch.vtensor<[1,56,?,?],f32>) -> !torch.vtensor<[1,28,?,?],f32> {
+  %int100 = torch.constant.int 100
+  %int1 = torch.constant.int 1
+  %int0 = torch.constant.int 0
+  %int2 = torch.constant.int 2
+  %0 = torch.aten.slice.Tensor %arg0, %int1, %int0, %int100, %int2 : !torch.vtensor<[1,56,?,?],f32>, !torch.int, !torch.int, !torch.int, !torch.int -> !torch.vtensor<[1,28,?,?],f32>
```

Comment:
Going by the [docs](https://mlir.llvm.org/docs/Dialects/TensorOps/#tensorextract_slice-tensorextractsliceop) for `tensor.extract_slice`, a reduction in rank is allowed. Does `aten.slice.Tensor` operate the same way, and if not, do we need to add an extra check to ensure we're lowering this consistent with the semantics of `aten.slice.Tensor`?

Reply:
We should confirm `aten.slice.Tensor` does not support this case. If so (which I'm assuming is the case), it seems enough to add a comment here documenting that and ensuring we don't create `tensor.extract_slice` with any mask that does rank reduction. 

Wdyt?

Reply:
> We should confirm aten.slice.Tensor does not support this case.

+1. An easy way to check PyTorch semantics is using python lit tests, starting with torch.nn.Module and converting to Torch dialect using the dynamo frontend. See [example](https://github.com/cruise-automation/mlir-tcp/blob/main/test/python/fx_import/basic_test.py). @zezhang can you please add this?

If we realize aten.slice.Tensor does allow rank reduction (similar to tensor.extract_slice), please ignore the following:

> ensuring we don't create tensor.extract_slice with any mask that does rank reduction.

I'm wondering about the case when a given set of offset+size+stride values that does _not_ incur rank reduction with `aten.slice.Tensor` but does so when translated to `tensor.extract_slice`, which seems to be when there's a leading dimension with 1 in the size and stride going by this example?
```
%1 = tensor.extract_slice %0[0, 0, 0][1, 16, 4][1, 1, 1] :
  tensor<8x16x4xf32> to tensor<16x4xf32>
```
Is there an option to disable rank reduction? If not, 1) having an explicit match failure during conversion, or 2) adding an expand shape to match semantics might be better.

Reply:
Good point. I have tried multiple examples, including both leading dim size of 1, and slice size of 1. Looks like the replaced `ExtractSliceOp` + `CastOp` never performs the rank reduction. This is becuase we have: